### DISCUSSION TITLE
Graph API Permissions - Global Admin?

### DISCUSSION THREAD

**Discussion by alzia-msft** [microsoft-search/pnp-modern-search#3199] [Q&A] [answered]
My client is asking for documentation if the PnP Modern Search web part install requires permission from a SPO or global admin. There are certain Graph APIs whose permission approvals are ideal for the web part performance -

<img width="354" alt="image" src="https://github.com/microsoft-search/pnp-modern-search/assets/92477952/91d9de5d-14e5-40c4-8f50-337038ce2cc5">

I understand the permissions of the Graph API are given in SPO admin center, howver permissions like Calendar.Read, Mail.Read, Chat.Read etc. are beyond the context of SPO, these are all APIs that run from EXO & Teams, can the blessings of an SPO admin be enough for the web part to function or we may need a global admin to provide these permissions? Since this is an opensource webpart, confirmation from the authors regarding the permissions would be appreciated. My client has a formal process to follow a security protocol, confirming if this requires additional permissions beyond SPO admin. I did not see any documentation elsewhere that mentions that SPO admin should be good enough for the permissions.

**Comment by alzia-msft** [comment]
@wobba for visbility.

**Comment by wobba** [comment, marked as answer]
You only have to grant the Graph scopes you need if using the Graph provider or the Graph Toolkit for displaying people cards. So yes, we request all which are used somewhere, but you choose to grant what you want/need.

**Reply by alzia-msft** [reply]
Thanks, @wobba! For the Graph scopes, would the global admin permission be required or SPO admin previlages will suffice?

**Reply by wobba** [reply]
If you can access the API page under SharePoint admin you have the needed rights. So I believe SharePoint admin rights is what is needed. 

https://<tenant>-admin.sharepoint.com/_layouts/15/online/AdminHome.aspx#/webApiPermissionManagement

**Reply by alzia-msft** [reply]
Thanks, @wobba! Appreciate the confirmation...
